File: practices/bad_update/b1.py
def dfs(node, graph, visited, crash):
    if node == crash:
        return
    visited[node] = True
    for nxt in graph[node]:
        if not visited[nxt] and nxt != crash:
            dfs(nxt, graph, visited, crash)

n, m = map(int, input().split())
graph = [[] for _ in range(n + 1)]

for _ in range(m):
    a, b = map(int, input().split())
    graph[a].append(b)
    graph[b].append(a)

max_fail = 0

# Try crashing each node except 1
for crash in range(2, n + 1):
    visited = [False] * (n + 1)
    dfs(1, graph, visited, crash)

    fail_count = 0
    for i in range(1, n + 1):
        if not visited[i]:
            fail_count += 1

    if fail_count > max_fail:
        max_fail = fail_count

print(max_fail)

File: practices/bad_update/b1.cpp
#include <iostream>
#include <vector>
using namespace std;

void dfs(int node, vector<vector<int>>& graph, vector<bool>& visited, int crash) {
    if (node == crash) return;
    visited[node] = true;
    for (int next : graph[node]) {
        if (!visited[next] && next != crash)
            dfs(next, graph, visited, crash);
    }
}

int main() {
    int n, m;
    cin >> n >> m;

    vector<vector<int>> graph(n + 1);
    for (int i = 0; i < m; i++) {
        int a, b;
        cin >> a >> b;
        graph[a].push_back(b);
        graph[b].push_back(a);
    }

    int maxFail = 0;

    for (int crash = 2; crash <= n; crash++) {
        vector<bool> visited(n + 1, false);
        dfs(1, graph, visited, crash);

        int failCount = 0;
        for (int i = 1; i <= n; i++) {
            if (!visited[i]) failCount++;
        }

        if (failCount > maxFail)
            maxFail = failCount;
    }

    cout << maxFail << endl;
    return 0;
}
